### PR TITLE
osu! custom changes (remaining)

### PR DESCRIPTION
**Pull request by peppy**
These are the changes we currently have applied in our custom fork of OpenTK.

Should be pretty easy to get most into upstream.

- Focus fix/hack isn't included. This didn't work well and should be fixed upstream in a better way.
- Targeting framework version isn't modified. I had it set to 4.x so I could build without having older .NET framework versions installed.

### CODE REVIEW

File: src/OpenTK/Math/Matrix2.cs [hunk 1]
```diff
@@ -350,6 +350,22 @@ public static void Mult(ref Matrix2 left, ref Matrix2 right, out Matrix2 result)
         }
 
         /// <summary>
+        /// Multiplies a matrix and a vector.
+        /// </summary>
+        /// <param name="left">The left operand of the multiplication.</param>
+        /// <param name="right">The right operand of the multiplication.</param>
+        /// <param name="result">A new instance that is the result of the multiplication.</param>
+        public static void Mult(ref Matrix2 left, ref Vector2 right, out Vector2 result)
```
Comment (Tom94) [inline src/OpenTK/Math/Matrix2.cs]: I don't think we use `Mult` anywhere.

File: src/OpenTK/Math/Vector2.cs [hunk 2]
```diff
@@ -667,6 +693,19 @@ public static void BaryCentric(ref Vector2 a, ref Vector2 b, ref Vector2 c, floa
         }
 
         /// <summary>
+        /// Transforms a vector by a Matrix rotation.
+        /// </summary>
+        /// <param name="vec">The vector to transform.</param>
+        /// <param name="mat">The matrix to rotate the vector by.</param>
+        /// <returns>The result of the operation.</returns>
+        public static Vector2 Transform(ref Matrix2 mat, ref Vector2 vec)
```
Comment (Tom94) [inline src/OpenTK/Math/Vector2.cs]: The 2 `Transform` funcs should become extension methods for now.

File: src/OpenTK/Math/Vector2.cs [hunk 3]
```diff
@@ -779,6 +830,28 @@ public static void Transform(ref Vector2 vec, ref Quaternion quat, out Vector2 r
         }
 
         /// <summary>
+        /// Transform a Vector by the given Matrix2
+        /// </summary>
+        /// <param name="mat">The desired transformation</param>
+        /// <param name="vec">The vector to transform</param>
+        /// <returns>The transformed vector</returns>
+        public static Vector2 operator *(Vector2 vec, Matrix2 mat)
```
Comment (Tom94) [inline src/OpenTK/Math/Vector2.cs]: These overloaded multiplications can be dropped entirely and in our codebase the `Transform` extensions should be used.

File: src/OpenTK/Math/Vector2.cs (hunk 2 above)
Comment (peppy) [reply src/OpenTK/Math/Vector2.cs]: sure about this? the same methods already exist in opentk for vector3, so i'd say having this here makes sense?

Comment (Tom94) [reply src/OpenTK/Math/Vector2.cs]: Depends on whether they have the same handedness or not. If they don't, then making this an extension method is the quickest and easiest way to get stuff into opentk upstream without us having to change our codebase. We can always change the handedness later on and then try to get 2D `Transform` into opentk.

Comment (peppy) [reply src/OpenTK/Math/Vector2.cs]: oh right, they are also back-to-front.

**Comment by peppy** [comment]
Closing as no longer relevant.

**Pull request closed without merging**